python/dgl/distributed/dist_graph.py

```diff
@@ -73,14 +73,15 @@ def _copy_graph_to_shared_mem(g, graph_name, graph_format):
     new_g.edata['inner_edge'] = _to_shared_mem(g.edata['inner_edge'],
                                                _get_edata_path(graph_name, 'inner_edge'))
     new_g.edata[EID] = _to_shared_mem(g.edata[EID], _get_edata_path(graph_name, EID))
+    new_g.edata[ETYPE] = _to_shared_mem(g.edata[ETYPE], _get_edata_path(graph_name, ETYPE))
     return new_g
 
 FIELD_DICT = {'inner_node': F.int32,    # A flag indicates whether the node is inside a partition.
               'inner_edge': F.int32,    # A flag indicates whether the edge is inside a partition.
               NID: F.int64,
               EID: F.int64,
-              NTYPE: F.int16,
-              ETYPE: F.int16}
+              NTYPE: F.int32,
+              ETYPE: F.int32}
```

Comment:
do you think int16 isn't large enough?

Reply:
There is a type mismatch (F.int16 vs. F.int32) between here and other places in the distributed module. For example in BasicPartitionBook.map_to_per_etype will return an int32 tensor storing etypes.